### ISSUE TITLE
bug: Test command tries to run non-test files

### ISSUE THREAD

**Issue by raspyweather** [ionic-team/stencil#3221]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.7.0

### Current Behavior

a component created with the name e.g. "link-test" gets detected as test file. 

### Expected Behavior

Only Test files (e2e & spec) should be detected as test files and executed

### Steps to Reproduce

1. Create a component named "link-test"
2. Run all tests
3. See following error message
4. 
```
FAIL  src/components/link-test/link-test.tsx
  ● Test suite failed to run

    Your test suite must contain at least one test.

      at onResult (node_modules/@jest/core/build/TestScheduler.js:175:18)
      at node_modules/@jest/core/build/TestScheduler.js:304:17
      at node_modules/emittery/index.js:260:13
          at Array.map (<anonymous>)
      at Emittery.Typed.emit (node_modules/emittery/index.js:258:23)
```

### Code Reproduction URL

https://github.com/raspyweather/stencil-issues

**Comment by rwaskiewicz** [comment]
Thanks! I've confirmed that this is a bug. This will affect any non-test file ending with:
- `test.tsx`
- `spec.tsx`
- `e2e.tsx`

(as well as files with the extensions `js`, `jsx`, `ts`). Examples include:

- `link-test.tsx`
- `my-component-spec.ts`
- `foo-e2e.js`
- etc.

As a workaround, I suggest folks rename their non-test files to not end in `test | spec | e2e` before the file extension.

I'm going to ingest this into our backlog for the team to fix

**Comment by rwaskiewicz** [comment]
This bug has been resolved in [Stencil v2.14.1](https://github.com/ionic-team/stencil/releases/tag/v2.14.1), which was released today (2022.03.07). I'm going to close this issue and mark it as shipped, thanks again!
